### PR TITLE
Add org.keystore_explorer.KeyStoreExplorer

### PR DESCRIPTION
**Pull request by ghazyami**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly. KeyStore Explorer is an open source, intuitive GUI replacement for the Java command-line utilities keytool and jarsigner that supports creating, managing and examining keystores, keys, certificates, certificate requests, certificate revocation lists and more.
- [x] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link: kaikramer/keystore-explorer#388**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: org.keystore_explorer.KeyStoreExplorer.metainfo.xml
Comment (bbhtt) [inline org.keystore_explorer.KeyStoreExplorer.metainfo.xml:1]: this should be submitted upstream https://github.com/kaikramer/keystore-explorer/

File: org.keystore_explorer.KeyStoreExplorer.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk
+command: kse
+rename-desktop-file: kse.desktop
+finish-args:
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --device=dri
+  - --filesystem=home
```
Comment (bbhtt) [inline org.keystore_explorer.KeyStoreExplorer.yml:14]: why do you need `home`? Does it support filechooser portals?

File: org.keystore_explorer.KeyStoreExplorer.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk/install.sh
+
+  - name: KeyStoreExplorer
+    buildsystem: simple
+    build-commands:
+      - mkdir -p deb-package
+      - ar p KeyStoreExplorer.deb data.tar.gz | tar -xzf - -C deb-package
+      - mv deb-package/opt/kse/* /app/
```
Comment (bbhtt) [inline org.keystore_explorer.KeyStoreExplorer.yml:27]:
```suggestion
      - bsdtar -Oxf KeyStoreExplorer.deb 'data.tar.gz' | bsdtar -xf -
      - mv opt/kse/* /app/
```

Comment (bbhtt) [reply org.keystore_explorer.KeyStoreExplorer.yml:27]: does this work?

File: org.keystore_explorer.KeyStoreExplorer.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      - mkdir /app/bin/
```
Comment (bbhtt) [inline org.keystore_explorer.KeyStoreExplorer.yml:28]:
```suggestion
      - mkdir -p /app/bin/
```

File: org.keystore_explorer.KeyStoreExplorer.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      - install -Dm644 /app/icons/kse_32.png /app/share/icons/hicolor/32x32/apps/org.keystore_explorer.KeyStoreExplorer.png
+      - install -Dm644 /app/icons/kse_16.png /app/share/icons/hicolor/16x16/apps/org.keystore_explorer.KeyStoreExplorer.png
+      - desktop-file-edit --set-icon=org.keystore_explorer.KeyStoreExplorer /app/kse.desktop
+      - desktop-file-install /app/kse.desktop --dir=/app/share/applications
+
+    sources:
+      - type: file
+        dest-filename: KeyStoreExplorer.deb
+        url: https://github.com/kaikramer/keystore-explorer/releases/download/v5.5.3/kse_5.5.3_all.deb
+        sha256: 59f3e161318ed717c2d717f38b0af1049a95e2eb7924e9ef6cf7ae3f6232f82b
```
Comment (bbhtt) [inline org.keystore_explorer.KeyStoreExplorer.yml:44]: why not build from source?

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 1 above)
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml:14]: - `home` is not explicitly needed, Downloads or Documents can work, the app main focus is to work with files (certificates, keys, etc..) so I though `home` is more suitable default. let me know if I should switcht to something else.
- It does not support filechooser portal.

Comment (bbhtt) [reply org.keystore_explorer.KeyStoreExplorer.yml:14]: If you can tighten it, it's better but certs and keys are often in project directories and not in a standard xdg-downloads directory or similar.

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 4 above)
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml:44]: This is a Java project that uses gradle as a build tool, it requires interne to download the dependencies during build, not sure if there is a way to do that or not.

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 2 above)
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml:27]: Works

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 4 above)
Comment (bbhtt) [reply org.keystore_explorer.KeyStoreExplorer.yml:43]: Have you tried https://github.com/flatpak/flatpak-builder-tools/tree/master/gradle#flatpak-gradle-generator ? This is an example that uses it https://github.com/flathub/org.ghidra_sre.Ghidra/blob/master/org.ghidra_sre.Ghidra.json

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 1 above)
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml]: switched to `xdg-documents/kse:create`

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 4 above)
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml:44]: I have not tried it, I will give it a go.

Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.yml:44]: I was not successfull
- I was able to build locally and get the log file, but parsing it with [flatpak-builder-tools - gradle](https://github.com/flatpak/flatpak-builder-tools/tree/master/gradle) did not work, complaint about some URL from the log.
- Also tried this one with no luck [jwharm/flatpak-gradle-generator](https://github.com/jwharm/flatpak-gradle-generator)

I would like to continue with the `.deb` package if that is not a blocker, after all it's pulled directly from upstream.

**Comment by bbhtt** [comment]
bot, build org.keystore_explorer.KeyStoreExplorer

File: org.keystore_explorer.KeyStoreExplorer.metainfo.xml
Comment (bbhtt) [reply org.keystore_explorer.KeyStoreExplorer.metainfo.xml:1]: can you submit this to upstream? This is useful or non Flatpak packages too it's a freedesktop standard.

File: org.keystore_explorer.KeyStoreExplorer.yml (hunk 4 above)
Comment (bbhtt) [reply org.keystore_explorer.KeyStoreExplorer.yml:44]: Ok

File: org.keystore_explorer.KeyStoreExplorer.metainfo.xml
Comment (ghazyami) [reply org.keystore_explorer.KeyStoreExplorer.metainfo.xml:1]: I created a ticket kaikramer/keystore-explorer#527

**Pull request closed without merging**